### ISSUE TITLE
strange error: cannot find value `var_i32` in this scope

### ISSUE THREAD

**Issue by Dushistov** [dtolnay/mashup#2]
My test code is:

```Rust
#![recursion_limit = "256"]
#[macro_use]
extern crate mashup;

macro_rules! foo {
    ($func:ident, [$( $Type:ident ),*]) => {{
        mashup! {
            $(
                m["var" $Type] = var_ $Type;
                )*
        }
        m! {
            $(
                let "var" $Type = $Type::default();
            )*
                f($("var" $Type),*);
        }
    }}
}

fn f(_: i32, _: f64) {}
fn main() {
    foo!(f, [i32, f64]);
}
```

The compiler error:

```
error[E0425]: cannot find value `var_i32` in this scope
  --> src/lib.rs:23:5
   |
23 |     foo!(f, [i32, f64]);
   |     ^^^^^^^^^^^^^^^^^^^^
   |     |
   |     not found in this scope
   |     in this macro invocation
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0425]: cannot find value `var_f64` in this scope
```

But if I run `cargo +nightly rustc -- -Z unstable-options --pretty=expanded`, I got:
```rust
        let var_i32 = i32::default();
        let var_f64 = f64::default();
        f(var_i32, var_f64);
```

So generated code is fine, and should not gives any compile time errors

**Comment by dtolnay** [comment]
Thanks! I released 0.1.3 with a fix. Your code should work as written after a `cargo update`.

**Comment by Dushistov** [comment]
Great! Thank you.
